static/flow/views/pi-selector-panel.js

```diff
@@ -481,14 +558,26 @@ var PiSelectorPanel = function(options) {
         else{
             dsDisplayedName = editor.getProgramEditorPanel().getDataStorageBlockDisplayedName();
             if(dsDisplayedName == ""){
-                alert("Please enter a valid dataset name on the data storage block.");
-                return;
+                modalAlert({
+                    title: 'Cannot Run Program', 
+                    message: "Please enter a valid dataset name on the data storage block.", 
+                    nextFunc: function() { 
+                        //update button
+                        updateProgramButtons(false, false, false);                    
+                    }});    
+                return;    
             }    
             
             var haveValidSequenceNames = editor.getProgramEditorPanel().validSequenceNames();
             if(!haveValidSequenceNames){
-                alert("Please enter a valid name for each type connected to the data storage block.");
-                return;
+                 modalAlert({
+                    title: 'Cannot Run Program', 
+                    message: "Please enter a valid name for each type connected to the data storage block.", 
+                    nextFunc: function() {    
+                        //update button
```

Comment:
Don't think any of these comments are necessary

Reply:
Agree